### ISSUE TITLE
Job run fail

### ISSUE THREAD

**Issue by ksoufan** [nautobot/nautobot-plugin-ssot-arista-cloudvision#116]
### Environment
* Python version:  3.10.6
* Nautobot version:  1.4.7
* aristacv-sync version:  1.4.0
* aristacvp version: 2022.3

Sync info from CVP to Nautobot

Failing with below trace
Traceback (most recent call last):
  File "/opt/nautobot/lib/python3.10/site-packages/nautobot_ssot/jobs/base.py", line 332, in run
    self.sync_data()
  File "/opt/nautobot/lib/python3.10/site-packages/nautobot_ssot/jobs/base.py", line 137, in sync_data
    self.load_source_adapter()
  File "/opt/nautobot/lib/python3.10/site-packages/nautobot_ssot_aristacv/jobs.py", line 117, in load_source_adapter
    self.source_adapter.load()
  File "/opt/nautobot/lib/python3.10/site-packages/nautobot_ssot_aristacv/diffsync/adapters/cloudvision.py", line 190, in load
    self.load_devices()
  File "/opt/nautobot/lib/python3.10/site-packages/nautobot_ssot_aristacv/diffsync/adapters/cloudvision.py", line 53, in load_devices
    self.load_interfaces(device=new_device)
  File "/opt/nautobot/lib/python3.10/site-packages/nautobot_ssot_aristacv/diffsync/adapters/cloudvision.py", line 62, in load_interfaces
    chassis_type = cloudvision.get_device_type(client=self.conn, dId=device.serial)
  File "/opt/nautobot/lib/python3.10/site-packages/nautobot_ssot_aristacv/utils/cloudvision.py", line 472, in get_device_type
    if query["fixedSystem"] is None:
KeyError: 'fixedSystem'

**Comment by jdrew82** [comment]
@ksoufan That's odd as I'm pretty sure the only options are fixed or chassis. Is it possible for you to get the results of the query `get_device_type` method? You should be able to get it by doing the following:

```
from django.conf import settings
PLUGIN_SETTINGS = settings.PLUGINS_CONFIG["nautobot_ssot_aristacv"]
from nautobot_ssot_aristacv.utils import cloudvision
from cloudvision.Connector.codec import Wildcard
from cloudvision.Connector.grpc_client.grpcClient import create_query
client = cloudvision.CloudvisionApi(
            cvp_host=PLUGIN_SETTINGS["cvp_host"],
            cvp_port=PLUGIN_SETTINGS.get("cvp_port", "8443"),
            verify=PLUGIN_SETTINGS["verify"],
            username=PLUGIN_SETTINGS["cvp_user"],
            password=PLUGIN_SETTINGS["cvp_password"],
            cvp_token=PLUGIN_SETTINGS["cvp_token"],
        )
dId = <string of your chassis device's serial>
pathElts = ["Sysdb", "hardware", "entmib"]
query = cloudvision.get_query(client, dId, pathElts)
query = unfreeze_frozen_dict(query)
from pprint import pprint
pprint(query)
```

**Comment by ksoufan** [comment]
Sorry, I'm not Django expert, but when trying your code getting error

django.core.exceptions.ImproperlyConfigured: Requested setting PLUGINS_CONFIG, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.

When I call  in python  >>setting.configure()   error change to
AttributeError: module 'django.conf.global_settings' has no attribute 'PLUGINS_CONFIG'. Did you mean: 'LOGGING_CONFIG'?

As for platform options, cloudvision allow to import Arista Access points and 3rd party using SNMP, which I did in my deployment, I think that may be the reason for the failure.

**Comment by jdrew82** [comment]
@ksoufan I apologize. I should have been clearer. You need to run those commands from the nautobot shell by either typing `nautobot-server shell_plus` in the Nautobot virtualenv on a regular VM or by doing the same inside the Nautobot container if using Docker.

**Comment by ksoufan** [comment]
Going into the shell_plus helped with the Django load, but the client plugin failing
client = cloudvision.CloudvisionApi(
            cvp_host=PLUGIN_SETTINGS["192.168.20.200"],
            cvp_port=PLUGIN_SETTINGS.get("443", "8443"),
            verify=PLUGIN_SETTINGS["verify"],
            username=PLUGIN_SETTINGS["kenan"],
            password=PLUGIN_SETTINGS["xxxxxxxx"],
            cvp_token=PLUGIN_SETTINGS[""],
        )
Traceback (most recent call last):
  File "<console>", line 2, in <module>
KeyError: '192.168.20.200'

**Comment by jdrew82** [comment]
@ksoufan It appears that you're trying to insert settings for CVP into the commands I gave you unnecessarily. The entire reason that I gave the commands I did is that it will pull the settings for your CVP instance directly from the PLUGIN_SETTINGS in your instance's `nautobot_config.py` file. You do not need to insert your instance settings again.

**Comment by ksoufan** [comment]
My bad.
Once I added the client got an error for freeze_dict, but it did print the query

NameError: name 'unfreeze_frozen_dict' is not defined
>>> pprint(query)
{'basePower': 156.0,
 'chassis': None,
 'fixedSystem': ['Sysdb', 'hardware', 'entmib', 'fixedSystem'],
 'hwMacAddr': 'fc:bd:67:0f:40:8d',
 'maxPorts': 65535,
 'name': 'entmib',
 'nextPhysicalIndex': <FrozenDict {'value': 2}>,
 'root': ['Sysdb', 'hardware', 'entmib', 'fixedSystem'],
 'systemMacAddr': 'fc:bd:67:0f:40:8d'}

However for other devices that such as Arista Access Point, got an update result
{}

**Comment by jdrew82** [comment]
See, that's strange cause your error was that `fixedSystem` doesn't exist in the query, but it's right there...

**Comment by jdrew82** [comment]
As for the `unfreeze_frozen_dict` method, I should have had `cloudvision.` in front of it as that's where the method resides.

**Comment by ksoufan** [comment]
Is it possible it getting the error when it try to check other devices (e.g access point) which does not have fixedSystem?

**Comment by jdrew82** [comment]
That was going to be my next question since you mentioned it in your response. When you do the query commands I gave you against an access point, does it return an empty dictionary?

**Comment by ksoufan** [comment]
Yes, it does return empty dictionary

**Comment by jdrew82** [comment]
In that case, we'll need to update the plugin to take those into account and ignore them until we can figure out how to process them. They most likely don't have the same endpoints that the switches and routers do. Unfortunately, I don't have any APs connected to CVP to do queries against to add them in at this time.

**Comment by ksoufan** [comment]
FYI. you could onboard device using SNMP, which return empty dictionary as well.

**Comment by jdrew82** [comment]
I'm currently working on a fix for #117 and I can include a fix for this with that PR since it should just be a simple check and log about ignoring the device. The PR should be in later today.
